src/core/utils/slaac_address.cpp

```diff
@@ -85,7 +85,7 @@ void Slaac::UpdateAddresses(otInstance *    aInstance,
             }
         }
 
-        if (!found)
```

Comment:
The intention here is to keep use the old address if its prefix is still valid in Network Data , other than generating new one; and remove it if its prefix is not valid on-mesh prefix any more.
Make sense?

Reply:
Yes, makes sense to me.
My problem is, when i perform the following steps in cli, no more slaac addresses are present without this change, even if i force a call to this function again:
  - ifconfig down
  - ifconfig up
  - thread start

But maybe the reason for that is to be found in a totally different place?!

Reply:
@marcoster, can you provide more detail on how to reproduce?

Using the latest master code, I was able to perform following:

```
$ ./output/x86_64-apple-darwin/bin/ot-cli-ftd 1
> panid 0x1234
Done
> ifconfig up
Done
> thread start
Done
> state
leader
Done
> ipaddr
fdde:ad00:beef:0:0:ff:fe00:fc00
fdde:ad00:beef:0:0:ff:fe00:ac00
fdde:ad00:beef:0:3ad2:cdc7:b98f:d273
fe80:0:0:0:5c40:b890:210e:b001
Done
> prefix add 2001:dead:beef:cafe::/64 paros
Done
> netdataregister
Done
> ipaddr
2001:dead:beef:cafe:73f6:154:c9f3:8fe6
fdde:ad00:beef:0:0:ff:fe00:fc00
fdde:ad00:beef:0:0:ff:fe00:ac00
fdde:ad00:beef:0:3ad2:cdc7:b98f:d273
fe80:0:0:0:5c40:b890:210e:b001
Done
> ifconfig down
Done
> ifconfig up
Done
> thread start
Done
> state
leader
Done
> ipaddr
2001:dead:beef:cafe:107:5d18:f714:6f3c
fdde:ad00:beef:0:0:ff:fe00:fc00
fdde:ad00:beef:0:0:ff:fe00:ac00
fdde:ad00:beef:0:3ad2:cdc7:b98f:d273
fe80:0:0:0:5c40:b890:210e:b001
Done
```

Note that the SLAAC addresses are added back properly.

Reply:
Sorry - I just took a look at the upstream code again - guess my problem is fixed by the addition of
*otIp6AddUnicastAddress(aInstance, address)* in line 119 - missed that.